### PR TITLE
The layer select command goes crazy when the layer name is wrong #782

### PR DESCRIPTION
**Pull request by ricardoffmartins**
Create a function to verify if a layer exists.
New exception LayersFailure that is thrown when the given name doesn't exist.
Change the Root ImportPath from ImportPath("") to ImportPath("/")

ref: layerSelectCommand

### CODE REVIEW

File: src/build/build.scala [hunk 1]
```diff
@@ -498,16 +498,24 @@ object LayerCli {
     layout    <- cli.layout
     baseLayer <- Layer.base(layout)
     schema    <- baseLayer.mainSchema
-    cli       <- cli.hint(LayerArg, ImportPath("/") :: schema.importTree(layout, true).getOrElse(Nil))
+    cli       <- cli.hint(LayerArg,  schema.importTree(layout, true).getOrElse(Nil))
```
Comment (propensive) [inline src/build/build.scala:501]: Why was the `/` layer (the "root" layer) removed from the suggestions?

File: src/model/ids.scala [hunk 2]
```diff
@@ -73,7 +73,7 @@ case class ModuleId(key: String) extends Key(msg"module")
 object ImportPath {
   implicit val msgShow: MsgShow[ImportPath] = ip => UserMsg(_.layer(ip.path))
   implicit val stringShow: StringShow[ImportPath] = _.path
-  val Root: ImportPath = ImportPath("")
+  val Root: ImportPath = ImportPath("/") //Can i change this?
```
Comment (propensive) [inline src/model/ids.scala:76]: I think so, though I would be interested to know why everything worked before, if the `Root` was incorrect...

**Review by propensive — APPROVED** [review]

File: src/build/build.scala (hunk 1 above)
Comment (ricardoffmartins) [reply src/build/build.scala:501]: It was added on ids.scala.

File: src/model/ids.scala (hunk 2 above)
Comment (ricardoffmartins) [reply src/model/ids.scala:76]: Maybe because the Root "/" was being added inside the select function. and it would appear in completions.

**Comment by ricardoffmartins** [comment]
@propensive 
there is also this 
https://github.com/ricardoffmartins/fury/blob/c33d3112acbcc2766479084953cfe78f0a4be484/src/model/ids.scala#L106

I didnt change this but is this correct?
(rawParts ++ xs) 
this looks like it could explain the weird path

**Pull request merged**